tests/integration/vpp/040_acl_test.go

```diff
@@ -0,0 +1,906 @@
+			t.Logf("Found modified ACL aclIPrules with aclName  %v", item.Meta.Tag)
+		}
+		if item.ACL.Interfaces.String() == "" {
+			t.Logf("Interface assignment was removed")
+		} else {
+			t.Fatalf("Interface assignment was not removed.")
+		}
+	}
+
+	err = h.DeleteACL(aclIdx)
+	Expect(err).To(BeNil())
+	t.Logf("deleting acls succeed")
+
+	acls, errx = h.DumpACL()
+	Expect(errx).To(BeNil())
+	aclCnt = len(acls)
+	Expect(aclCnt).Should(Equal(0))
+	t.Logf("%d acls dumped", aclCnt)
+
+}
+
+// Test add MACIP acl rules
+func TestCRUDMacIPAcl(t *testing.T) {
+	ctx := setupVPP(t)
+	defer ctx.teardownVPP()
+
+	ih := ifplugin_vppcalls.CompatibleInterfaceVppHandler(ctx.vppBinapi, logrus.NewLogger("test"))
+
+	const ifName = "loop1"
+	ifIdx, errI := ih.AddLoopbackInterface(ifName)
+	Expect(errI).To(BeNil())
+	t.Logf("interface created %v", ifIdx)
+
+	const ifName2 = "loop2"
+	ifIdx2, errI2 := ih.AddLoopbackInterface(ifName2)
+	Expect(errI2).To(BeNil())
+	t.Logf("interface created %v", ifIdx2)
+
+	ifIndexes := ifaceidx.NewIfaceIndex(logrus.NewLogger("test-iface1"), "test-iface1")
+	ifIndexes.Put(ifName, &ifaceidx.IfaceMetadata{
+		SwIfIndex: ifIdx,
+	})
+	ifIndexes.Put(ifName2, &ifaceidx.IfaceMetadata{
+		SwIfIndex: ifIdx2,
+	})
+
+	h := aclplugin_vppcalls.CompatibleACLVppHandler(ctx.vppBinapi, ifIndexes, logrus.NewLogger("test"))
+	if h == nil {
+		t.Fatalf("handler was not created")
+	}
+
+	acls, errx := h.DumpMACIPACL()
+	Expect(errx).To(BeNil())
+	aclCnt := len(acls)
+	Expect(aclCnt).Should(Equal(0))
+	t.Logf("no acls dumped")
+
+	const aclname = "test6"
+	aclIdx, err := h.AddMACIPACL(aclMACIPrules, aclname)
+	Expect(err).To(BeNil())
+	Expect(aclIdx).To(BeEquivalentTo(0))
+	t.Logf("acl added - with index %d", aclIdx)
+
+	err = h.SetMACIPACLToInterfaces(aclIdx, []uint32{ifIdx})
+	Expect(err).To(BeNil())
+	t.Logf("acl with index %d was assigned to interface %v", aclIdx, ifName)
+
+	acls, errx = h.DumpMACIPACL()
+	Expect(errx).To(BeNil())
+	aclCnt = len(acls)
+	Expect(aclCnt).Should(Equal(1))
+	t.Logf("%d acls dumped", aclCnt)
+
+	var rules []*acl.ACL_Rule
+	var isPresent bool
+	var isForInterface bool
+	for _, item := range acls {
+		rules = item.ACL.Rules
+		if (item.Meta.Index == aclIdx) && (aclname == item.Meta.Tag) {
+			t.Logf("Found ACL aclMACIPrules with aclName  %v", item.Meta.Tag)
+			for _, rule := range rules {
+				if (rule.MacipRule.SourceAddress == "192.168.0.1") &&
+					(rule.MacipRule.SourceAddressPrefix == 16) &&
+					(strings.ToLower(rule.MacipRule.SourceMacAddress) == strings.ToLower("11:44:0A:B8:4A:35")) &&
+					(rule.MacipRule.SourceMacAddressMask == "ff:ff:ff:ff:00:00") {
+					isPresent = true
+					break
+				}
+			}
+			// check assignation to interface
+			t.Logf("%v", item)
+			t.Logf("%v", item.ACL.Interfaces)
+			for _, intf := range item.ACL.Interfaces.Ingress {
+				if intf == ifName {
+					isForInterface = true
+					break
+				}
+			}
+		}
+	}
+	if !isPresent {
```

Comment:
You can simplify this check using [`Annotated Assertions`](https://onsi.github.io/gomega/#annotating-assertions), like this:

```go
Expect(isPresent).To(BeFalse(), "Configured IP should be present")
```

Reply:
Again thanks for advice